test/partiql-tests-runner/src/main/kotlin/org/partiql/runner/ReportAnalyzer.kt

```diff
@@ -0,0 +1,153 @@
+    private val firstIgnoreSize = first.ignoredSet.size
+    private val secondPassingSize = second.passingSet.size
+    private val secondFailingSize = second.failingSet.size
+    private val secondIgnoreSize = second.ignoredSet.size
+
+    private val firstTotalSize = firstPassingSize + firstFailingSize + firstIgnoreSize
+    private val secondTotalSize = secondPassingSize + secondFailingSize + secondIgnoreSize
+
+    private val firstPassingPercent = firstPassingSize.toDouble() / firstTotalSize * 100
+    private val secondPassingPercent = secondPassingSize.toDouble() / secondTotalSize * 100
+
+    private val firstNameShort = "$BASE (${first.engine.uppercase()}-${first.commitIdShort.uppercase()})"
+    private val secondNameShort = "$TARGET (${second.engine.uppercase()}-${second.commitIdShort.uppercase()})"
+
+    fun generateComparisonReport(limit: Int): String {
+        return buildString {
+            appendTitle(this)
+            appendTable(this)
+            appendSummary(this)
+            appendOptionalNowFailingTests(this, limit)
+            appendOptionalNowPassingTests(this, limit)
+        }
+    }
+
+    private fun appendTitle(out: Appendable) {
+        val icon = if (passingFirstFailingSecond.isEmpty()) ICON_CHECK else ICON_X
+        out.appendMarkdown("# $reportTitle $icon")
+    }
+
+    private fun appendTable(out: Appendable) {
+        out.appendLine("| | $firstNameShort | $secondNameShort | +/- |")
+        out.appendLine("| --- | ---: | ---: | ---: |")
+        out.appendLine(tableRow("% Passing", firstPassingPercent, secondPassingPercent))
+        out.appendLine(tableRow("Passing", firstPassingSize, secondPassingSize, true))
+        out.appendLine(tableRow("Failing", firstFailingSize, secondFailingSize, false))
+        out.appendLine(tableRow("Ignored", firstIgnoreSize, secondIgnoreSize, false))
+        out.appendLine(tableRow("Total Tests", firstTotalSize, secondTotalSize, true))
+    }
+
+    private fun tableRow(name: String, first: Double, second: Double): String {
+        val firstString = "%.2f".format(first)
+        val secondString = "%.2f".format(second)
+        val delta = second - first
+        val deltaIcon = getIconForComparison(first, second, true)
+        val deltaString = "%.2f".format(delta)
+        return "| $name | $firstString% | $secondString% | $deltaString% $deltaIcon |"
+    }
+
+    private fun tableRow(name: String, first: Int, second: Int, positiveDeltaGood: Boolean): String {
+        val delta = second - first
+        val deltaIcon = getIconForComparison(first, second, positiveDeltaGood)
+        return "| $name | $first | $second | $delta $deltaIcon |"
+    }
+
+    private fun <T> getIconForComparison(first: Comparable<T>, second: T, positiveDeltaGood: Boolean): String {
+        val comparison = first.compareTo(second)
+        return when {
+            comparison < 0 -> if (positiveDeltaGood) ICON_CHECK else ICON_CIRCLE_RED
+            comparison == 0 -> ICON_CHECK
```

Comment:
nit: wonder if for if there's no diff, we want some neutral indication consider :heavy_minus_sign: (`:heavy_minus_sign:`). ✅ would seem to indicate some good change but perhaps no regression is good?

Reply:
Yeah, I played with doing a grey icon, but it started to look a bit cluttered visually. I figured green/red are pretty easy indicators for reviewers that something is good/bad.